src/storage-controller/src/lib.rs

```diff
@@ -1892,122 +1887,129 @@ where
         }
 
         let mut updated_frontiers = None;
-        match self.stashed_response.take() {
-            None => (),
-            Some(StorageResponse::FrontierUppers(updates)) => {
-                self.update_write_frontiers(&updates);
-                updated_frontiers = Some(Response::FrontierUpdates(updates));
-            }
-            Some(StorageResponse::DroppedId(id)) => {
-                tracing::debug!("DroppedId for collection {id}");
-
-                if let Some(_collection) = self.collections.remove(&id) {
-                    // Nothing to do, we already dropped read holds in
-                    // `drop_sources_unvalidated`.
-                } else if let Some(export) = self.exports.get_mut(&id) {
-                    // TODO: Current main never drops export state, so we
-                    // also don't do that, because it would be yet more
-                    // refactoring. Instead, we downgrade to the empty
-                    // frontier, which satisfies StorageCollections just as
-                    // much.
-                    tracing::info!("downgrading read hold of export {id} to empty frontier!");
-                    export
-                        .read_hold
-                        .try_downgrade(Antichain::new())
-                        .expect("must be possible");
-                } else {
-                    soft_panic_or_log!(
-                        "DroppedId for ID {id} but we have neither ingestion nor export \
-                         under that ID"
-                    );
+
+        // Take the allocation so that we can call mut receiver functions in the loop. We put it
+        // back at the end of the loop.
+        let mut stashed_responses = std::mem::take(&mut self.stashed_responses);
```

Comment:
Maybe we shouldn't? If we drop the allocation instead, we don't have to worry about it ever getting large. Keeping it lets us safe some allocation overhead, but is that really relevant in this code?

Reply:
>  but is that really relevant in this code?

It's not, you're right. Switching to the simpler version